openmetadata-service/src/main/java/org/openmetadata/service/secrets/SecretsManager.java

```diff
@@ -51,11 +51,16 @@ public abstract class SecretsManager {
   public record SecretsConfig(
       String clusterName, String prefix, List<String> tags, Parameters parameters) {}
 
+  protected record SecretsIdConfig(
+      String separator,
+      Boolean needsStartingSeparator,
+      String cleanSecretReplacer,
+      Pattern secretIdPattern) {}
+
   @Getter private final SecretsConfig secretsConfig;
   @Getter private final SecretsManagerProvider secretsManagerProvider;
+  @Getter private final SecretsIdConfig secretsIdConfig;
```

Comment:
Can this and the buld method be in the Azure class since it seems to be Azure related ?

Reply:
kept it in the general one as as default with the idea to give room for any other SM to override it. Right now it's only Azure KV, but I am guessing other solutions might have their own specific conventions. The first implementation was a bit too AWS-centric